### PR TITLE
fix: user not able to enter space manually

### LINKED ISSUE #1508: Unable to type spaces when entering secret key
Version: 2.12.4

When trying to enter a secret key on the screen below by manually typing it, I am unable to type a [space] between any of the words.

image

To reproduce:

1. Click Sign in wtih Secret Key
2. Try to manually type a key using spaces

Copy and paste works as expected.

### PR DESCRIPTION
**Pull request by beguene**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1090385798).

Remove trim on input change

### CODE REVIEW

File: src/common/hooks/auth/use-sign-in.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export function useSignIn() {
 
   const handleSetSeed = useCallback(
     async (value: string) => {
-      const trimmed = value.trim();
+      const trimmed = value;
```
Comment (aulneau) [inline src/common/hooks/auth/use-sign-in.ts]: This change makes a few other parts of the code different now.

I'd actually recommend keeping this function as-is and having `onPaste` use this, and then create a different method for `onChange` (which is what would be used for someone manually typing the seed phrase)

**Review by aulneau — CHANGES_REQUESTED** [review]

File: src/common/hooks/auth/use-sign-in.ts (hunk 1 above)
Comment (beguene) [reply src/common/hooks/auth/use-sign-in.ts]: Sure !

Comment (beguene) [reply src/common/hooks/auth/use-sign-in.ts]: @aulneau To avoid code duplication, I added an option to trim since the behaviour is really similar

**Review by aulneau — APPROVED** [review]

**Comment by aulneau** [comment]
@beguene i think we still need a check at some point to trim it before submitting. What happens if someone types a seed phrase and adds a bunch of extra spaces to it at the end?

**Comment by beguene** [comment]
> @beguene i think we still need a check at some point to trim it before submitting. What happens if someone types a seed phrase and adds a bunch of extra spaces to it at the end?

Yes I agree we can improve. Can we use this PR as a quick fix since user anyway cannot currently entered any space ? I will add another PR to improve based on your comment.

**Comment by aulneau** [comment]
> > @beguene i think we still need a check at some point to trim it before submitting. What happens if someone types a seed phrase and adds a bunch of extra spaces to it at the end?
> 
> Yes I agree we can improve. Can we use this PR as a quick fix since user anyway cannot currently entered any space ? I will add another PR to improve based on your comment.

I mean we should QA it to ensure the removal of trim doesn't create new bugs

**Pull request merged**